### ISSUE TITLE
Consider adding unused imports removal project.

### ISSUE THREAD

**Issue by mshima** [jhipster/prettier-java#662]
I've created https://github.com/mshima/java-lint to remove unused imports in generator-jhipster generated code.
Would be nice if it could be moved to jhipster organization in java-parser/prettier-java repository.

**Comment by murdos** [comment]
I think there have already been some discussions about how far prettier-java should go with imports, in https://github.com/jhipster/prettier-java/issues/201.
The conclusion - as I understand it - was that prettier is a formatter, but should not act as a linter. So this might be out of the scope of prettier.

**Comment by mshima** [comment]
Yes, I understand that it should be not be implemented in prettier-plugin-java.
The proposal is to add a new package in this repository.

**Comment by DanielFran** [comment]
@clementdessoude Any thoughts on that?
@mshima Since the development is already done, working and used in the generator, can you provide a PR so we can evaluate the impacts?

**Comment by pascalgrimaud** [comment]
I agree with @murdos : this might be out of the scope of prettier

@mshima : if you want to include this project to jhipster organization, it should be at github.com/jhispter/java-lint
Just send an email to the mailing to discuss about it

**Comment by mshima** [comment]
> I agree with @murdos : this might be out of the scope of prettier
> 
> @mshima : if you want to include this project to jhipster organization, it should be at github.com/jhispter/java-lint Just send an email to the mailing to discuss about it

I only have interest to move to the `java-parser` repository to keep `java-lint` updated with it.
I don't have interest to move to jhipster organization if the structure will be kept the sample.

**Comment by mshima** [comment]
> can you provide a PR so we can evaluate the impacts?

@DanielFran the idea is to move the entire project to https://github.com/jhipster/prettier-java/tree/main/packages/java-lint.
As impact, maybe we should migrate prettier-java to vitest to match java-lint.
